### PR TITLE
Issue #10625: Fixed InputEqualsAvoidNullTextBlocks.java and removed it from linelength suppression

### PR DESCRIPTION
**Pull request by AbhinavRajputEXE**
#10625

### CODE REVIEW

**Comment by AbhinavRajputEXE** [comment]
@romani 
I made the needed changes in this one only, I tried making other many changes too, to make it go way under the 100 char. limit but it was giving me few errors, so I thought to do few things at a time and gradually increase it, that's why I didn't make all the changes suggested by you in this one, but I'll use those suggestions in other files where those files are exceeding the limit by so much.
please review this PR, whether these changes are correct or not.

**Review by romani — APPROVED** [review]
Yes, do updates in small batches.

Changing indentation, might resolve most problems.

**Review by romani — CHANGES_REQUESTED** [review]
Let's try to fix indentation. In this case violation message could be less cryptic.

As person who  reads such input files, it is extremely convenient to have message close to line with bad code.

**Comment by AbhinavRajputEXE** [comment]
@romani I change the indentation to +2 from +4 but it is giving me these errors now on running `mvn clean verify`
```
[ERROR] Failures: 
[ERROR]   EqualsAvoidNullCheckTest.testEqualsAvoidNullTextBlocks:241->AbstractModuleTestSupport.verifyWithInlineConfigParser:271->AbstractModuleTestSupport.verify:
287->AbstractModuleTestSupport.verify:325->AbstractModuleTestSupport.verify:347->AbstractModuleTestSupport.verify:377->AbstractModuleTestSupport.lambda$verify$1:38
0 Violations for C:\Users\abhin\Desktop\Checkstyle Folder\checkstyle\src\test\resources-noncompilable\com\puppycrawl\tools\checkstyle\checks\coding\equalsavoidnull
\InputEqualsAvoidNullTextBlocks.java differ.
missing (4)
#1      : 13:28: String literal expressions should be on the left side of an equals comparison.
#2      : 15:28: String literal expressions should be on the left side of an equals comparison.
#3      : 22:25: String literal expressions should be on the left side of an equals comparison.
#4      : 32:39: String literal expressions should be on the left side of an equalsIgnoreCase comparison.

unexpected (4)
#1      : 13:24: String literal expressions should be on the left side of an equals comparison.
#2      : 15:24: String literal expressions should be on the left side of an equals comparison.
#3      : 22:19: String literal expressions should be on the left side of an equals comparison.
#4      : 32:31: String literal expressions should be on the left side of an equalsIgnoreCase comparison.
---
expected: [13:28: String literal expressions should be on the left side of an equals comparison., 15:28: String literal expressions should be on the left side of a
n equals comparison., 22:25: String literal expressions should be on the left side of an equals comparison., 32:39: String literal expressions should be on the lef
t side of an equalsIgnoreCase comparison.]
but was : [13:24: String literal expressions should be on the left side of an equals comparison., 15:24: String literal expressions should be on the left side of a
n equals comparison., 22:19: String literal expressions should be on the left side of an equals comparison., 32:31: String literal expressions should be on the lef
t side of an equalsIgnoreCase comparison.]
[INFO]
[ERROR] Tests run: 3655, Failures: 1, Errors: 0, Skipped: 1
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------

```

**Comment by romani** [comment]
Look at https://github.com/checkstyle/checkstyle/blob/c374c8c7b68678b6648f6ba44d70364350dcd32e/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheckTest.java#L38 it contains column numbers in test methods, you need to update them also in that file

**Comment by AbhinavRajputEXE** [comment]
@romani 
I updated the column numbers.

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/equalsavoidnull/InputEqualsAvoidNullTextBlocks.java [hunk 1]
```diff
@@ -9,37 +9,37 @@
...
 
 public class InputEqualsAvoidNullTextBlocks {
-    public void equalsAvoid(String myString) {
-        if (myString.equals("stuff")) { // violation 'String literal expressions should be on the left side of an equals comparison.'
-        } // violation below 'String .* left .* of .* equals'
-        if (myString.equals("""
-                stuff""")) {
-        }
+  public void equalsAvoid(String myString) {
+    if (myString.equals("stuff")) { // violation 'String.*left.*of.*equals'
```
Comment (romani) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/equalsavoidnull/InputEqualsAvoidNullTextBlocks.java:13]: Your string is 75 characters long.

Can you reuse 25 characters to keep message most meaningful part

**Review by romani — CHANGES_REQUESTED** [review]
Items:

**Comment by AbhinavRajputEXE** [comment]
oh, I actually changed that, I don't know why it is not showing up in this commit, 
let me do it again.

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/equalsavoidnull/InputEqualsAvoidNullTextBlocks.java (hunk 1 above)
Comment (AbhinavRajputEXE) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/equalsavoidnull/InputEqualsAvoidNullTextBlocks.java]: I've made a few changes, are they correct?

Comment (AbhinavRajputEXE) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/equalsavoidnull/InputEqualsAvoidNullTextBlocks.java]: @romani 
please verify these new changes

**Review by romani — APPROVED** [review]
Thanks a lot

**Pull request merged**